src/python_testing/TC_AVSUM_2_4.py

```diff
@@ -0,0 +1,211 @@
+            TestStep(12, "Move to the newly saved preset"),
+            TestStep(13, "Verify that the MPTZPosition is that of the preset"),
+        ]
+        return steps
+
+    def pics_TC_AVSUM_2_4(self) -> list[str]:
+        pics = [
+            "AVSUM.S",
+        ]
+        return pics
+
+    @run_if_endpoint_matches(has_cluster(Clusters.CameraAvSettingsUserLevelManagement))
+    async def test_TC_AVSUM_2_4(self):
+        cluster = Clusters.Objects.CameraAvSettingsUserLevelManagement
+        attributes = cluster.Attributes
+        tilt_min_dut = tilt_max_dut = pan_min_dut = pan_max_dut = zoom_max_dut = None
+
+        self.step(1)  # Already done, immediately go to step 2
+
+        feature_map = await self.read_avsum_attribute_expect_success(attribute="FeatureMap")
+        self.has_feature_dptz = (feature_map & cluster.Bitmaps.Feature.kDigitalPTZ) != 0
+        self.has_feature_mpan = (feature_map & cluster.Bitmaps.Feature.kMechanicalPan) != 0
+        self.has_feature_mtilt = (feature_map & cluster.Bitmaps.Feature.kMechanicalTilt) != 0
+        self.has_feature_mzoom = (feature_map & cluster.Bitmaps.Feature.kMechanicalZoom) != 0
+        self.has_feature_mpresets = (feature_map & cluster.Bitmaps.Feature.kMechanicalPresets) != 0
+
+        # Get the Max and Min values
+        if self.has_feature_mzoom:
+            zoom_max_dut = await self.read_avsum_attribute_expect_success(attribute="ZoomMax")
+
+        if self.has_feature_mtilt:
+            tilt_min_dut = await self.read_avsum_attribute_expect_success(attribute="TiltMin")
+            tilt_max_dut = await self.read_avsum_attribute_expect_success(attribute="TiltMax")
+
+        if self.has_feature_mpan:
+            pan_min_dut = await self.read_avsum_attribute_expect_success(attribute="PanMin")
+            pan_max_dut = await self.read_avsum_attribute_expect_success(attribute="PanMax")
+
+        attribute_list = await self.read_avsum_attribute_expect_success(attribute="AttributeList")
+
+        # If Presets is empty, change MPTZPosition to some new value, save that as a Preset.  Change to a different value.
+        # Check that.
+        # Move to a discovered preset, or the created Preset.  Verify content.
+        self.step(2)
+        if self.has_feature_mpresets:
+            self.step(3)
+            asserts.assert_in(attributes.MaxPresets.attribute_id, attribute_list,
+                              "MaxPresets attribute is a mandatory attribute if MPRESETS.")
+            max_presets_dut = await self.read_avsum_attribute_expect_success(attribute="MaxPresets")
+
+            self.step(4)
+            asserts.assert_in(attributes.MPTZPresets.attribute_id, attribute_list,
+                              "MPTZPresets attribute is a mandatory attribut if MPRESETS.")
+
+            mptz_presets_dut = await self.read_avsum_attribute_expect_success(attribute="MPTZPresets")
+
+            self.step(5)
+            if mptz_presets_dut:
```

Comment:
More of a spec / impl question, but are these normally something the DUT sets or the client? ie, are you likely to receive anything in here if you don't set anything?

Reply:
General expectation is that presets exist because the client (or some non-Matter client, and stuff is getting reflected to us) set them.  Generally I probably would expect a DUT to be pre-loaded with presets.  I'm trying to handle either way though, if they're present, great, if not, we'll populate some and ensure it doesn't screw up.